### PR TITLE
change all types.FunctionType to callable

### PR DESCRIPTION
**Pull request by zjjott**

### CODE REVIEW

**Comment by merlintang** [comment]
@zjjott thanks for your pr, it is a good point. can you fix the test as well.

**Comment by zjjott** [comment]
@merlintang done, so sad to write this unittest assert

File: couler/tests/workflow_basic_test.py [hunk 1]
```diff
@@ -209,8 +226,26 @@ def test_workflow_config(self):
...
+        flip_coin()
+        for func_name in func_names:
+            if func_name == "self":
+                couler.set_exit_handler(couler.WFStatus.Succeeded, instance)
+            else:
+                couler.set_exit_handler(
+                    couler.WFStatus.Succeeded, getattr(instance, func_name)
+                )
+            couler.set_exit_handler(couler.WFStatus.Failed, tails)
+            # self.check_argo_yaml("workflow_basic_golden.yaml")
```
Comment (terrytangyuan) [inline couler/tests/workflow_basic_test.py:250]: Why is this commented out?

Comment (zjjott) [reply couler/tests/workflow_basic_test.py]: i forgot , fixed it

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**